client/state/notices/account-recovery/index.js

```diff
@@ -6,9 +6,43 @@ import { translate } from 'i18n-calypso';
 /**
  * Internal dependencies
  */
-import { errorNotice } from 'state/notices/actions';
+import {
+	successNotice,
+	errorNotice,
+} from 'state/notices/actions';
+
+import {
+	getAccountRecoveryPhone,
+	getAccountRecoveryEmail,
+} from 'state/account-recovery/settings/selectors';
+
 import { dispatchError } from '../utils';
 
+const getUpdateSuccessMessage = ( target, getState ) => {
+	switch ( target ) {
+		case 'phone':
+			const oldPhone = getAccountRecoveryPhone( getState() );
+
+			if ( null == oldPhone ) {
+				return translate( 'Successfully added new recovery SMS number.' );
+			}
+
+			return translate( 'Successfully updated recovery SMS number.' );
+
+		case 'email':
+			const oldEmail = getAccountRecoveryEmail( getState() );
+
+			if ( '' === oldEmail ) {
```

Comment:
[I left a note here](https://github.com/Automattic/wp-calypso/commit/c714c169d56c82abc20b72cafed7a13436dd5b79#commitcomment-20342783) on `null` vs empty strings, copying in case the commit disappears:

> Others may disagree here, but I'm not convinced an empty string is a good way to represent the absence of data. In places where you might need to check if the email address is missing, you'd need to remember to do `=== ''` or `.length === 0`, because `== null` would return `false`. This seems inconsistent with other selectors, like checking if an object is set.
> 
> Defaulting to an empty string seems like something that should be left as late as possible, for example in the view. That way the view can decide whether to display an empty string or something like "No email address".